packages/salesforcedx-vscode-core/src/commands/functions/forceFunctionInvoke.ts

```diff
@@ -8,31 +8,60 @@
 /**
  * Executes sfdx run:function --url http://localhost:8080 --payload=@functions/MyFunction/payload.json
  */
-import {
-  Command,
-  SfdxCommandBuilder
-} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
 import { Uri } from 'vscode';
+import { channelService, OUTPUT_CHANNEL } from '../../channels';
 import { nls } from '../../messages';
 import { notificationService } from '../../notifications';
 import { telemetryService } from '../../telemetry';
+import { OrgAuthInfo } from '../../util';
 import {
   FilePathGatherer,
   SfdxCommandlet,
-  SfdxCommandletExecutor,
   SfdxWorkspaceChecker
 } from '../util';
 import { FunctionService } from './functionService';
 
-export class ForceFunctionInvoke extends SfdxCommandletExecutor<string> {
-  public build(payloadUri: string): Command {
-    return new SfdxCommandBuilder()
-      .withDescription(nls.localize('force_function_invoke_text'))
-      .withArg('run:function')
-      .withFlag('--url', 'http://localhost:8080')
-      .withFlag('--payload', `@${payloadUri}`)
-      .withLogName('force_function_invoke')
-      .build();
+import { runFunction } from '@heroku/functions-core';
+import { LibraryCommandletExecutor } from '@salesforce/salesforcedx-utils-vscode/out/src';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+export class ForceFunctionInvoke extends LibraryCommandletExecutor<string> {
+  constructor(debug: boolean = false) {
+    super(
+      nls.localize('force_function_invoke_text'),
+      debug ? 'force_function_debug_invoke' : 'force_function_invoke',
+      OUTPUT_CHANNEL
+    );
+    if (debug) {
```

Comment:
@YousefHaggyHeroku - Not sure if I understand this. We need language based telemetry for both debug and regular invokes. What does this do?

Reply:
I'm not entirely sure, but right now in forceFunctionInvoke we only use language based telemetry on debug invokes: https://github.com/forcedotcom/salesforcedx-vscode/blob/158d4794e1a9498c13508b21809848c953fe7588/packages/salesforcedx-vscode-core/src/commands/functions/forceFunctionInvoke.ts#L72-L75

I tried to adapt closely to what already existed. I *think* we only log language based telemetry in the debug invoke because we don't assume there are any `FunctionService.startedExecutions` with the regular invoke. The regular invoke can technically be called on a function running in CLI directly, not VS Code. @randi274, you  probably have more context on this than I do, curious to hear your thoughts!

Reply:
Roger that, making that change now